lib/console/deployments/observability/webhook.ex

```diff
@@ -54,6 +54,32 @@ defmodule Console.Deployments.Observability.Webhook do
     |> ok()
   end
 
+  def payload(%ObservabilityWebhook{type: :datadog}, payload) do
+    case Datadog.normalize_datadog_payload(payload) do
+      [] -> {:error, "invalid payload"}
+      alerts ->
+        Enum.map(alerts, fn alert ->
+          Map.merge(%{
+            type: :datadog,
+            fingerprint: Map.get(alert, "id"),
+            annotations: Map.get(alert, "meta") || %{},
+            state: Datadog.state(
+              Map.get(alert, "status") ||
+              Map.get(alert, "alert_transition") ||
+              ""
+            ),
+            severity: Datadog.severity(alert),
+            url: Map.get(alert, "link") || Map.get(alert, "url") || "",
+            title: Map.get(alert, "title") || "Datadog Alert",
+            message: Datadog.summary(alert),
+            tags: Datadog.datadog_tags(alert),
+          }, add_associations(Datadog, alert))
+          |> backfill_raw()
```

Comment:
this is doing the regex matching you're doing internally in the datadog implementation (why i suspect it's duplicate work)